distributed/comm/tests/test_comms.py

```diff
@@ -457,6 +456,39 @@ async def test_inproc_specific_different_threads():
     await check_inproc_specific(run_coro_in_thread)
 
 
+@pytest.mark.asyncio
+async def test_inproc_continues_listening_after_handshake_error():
+    async def handle_comm():
+        pass
+
+    async with listen("inproc://", handle_comm) as listener:
+        addr = listener.listen_address
+        scheme, loc = parse_address(addr)
+        connector = get_backend(scheme).get_connector()
+
+        comm = await connector.connect(loc)
+        await comm.close()
+
+        comm = await connector.connect(loc)
+        await comm.close()
+
+
+@pytest.mark.asyncio
+async def test_inproc_handshakes_concurrently():
+    async def handle_comm():
+        pass
+
+    async with listen("inproc://", handle_comm) as listener:
+        addr = listener.listen_address
+        scheme, loc = parse_address(addr)
+        connector = get_backend(scheme).get_connector()
+
+        comm1 = await connector.connect(loc)
+        comm2 = await connector.connect(loc)
+        await comm1.close()
+        await comm2.close()
```

Comment:
This is not actually concurrently, is it?

Reply:
I was missing the context of the handshake. These connect attempts are happening concurrently since the connection is only properly established once the handshake is done. Since neither of these comms engages in the handshake, they are happening concurrently

Reply:
although the connections are opened sequentially the connections remain open concurrently